src/components/media-preview.cjsx

```diff
@@ -0,0 +1,158 @@
+React = require 'react'
+LoadableItem = require './loadable-item'
+TutorPopover = require './tutor-popover'
+
+_ = require 'underscore'
+S = require '../helpers/string'
+
+{MediaStore} = require '../flux/media'
+{ReferenceBookPageStore, ReferenceBookPageActions} = require '../flux/reference-book-page'
+
+
+MediaPreview = React.createClass
+  displayName: 'MediaPreview'
+  getInitialState: ->
+    popped: false
+    stick: false
+    media: null
+
+  propTypes:
+    mediaId: React.PropTypes.string.isRequired
+    bookHref: React.PropTypes.string.isRequired
+    cnxId: React.PropTypes.string.isRequired
+    mediaDOMOnParent: React.PropTypes.object
+    windowImpl: React.PropTypes.object
+    buffer: React.PropTypes.number
+    shouldLinkOut: React.PropTypes.bool
+
+  getDefaultProps: ->
+    buffer: 160
+    shouldLinkOut: false
+    windowImpl: window
+
+  componentWillMount: ->
+    {mediaId, cnxId} = @props
+    media = MediaStore.get(mediaId)
+    @updateMedia(media) if media?
+
+    unless media? or ReferenceBookPageStore.isLoading(cnxId) or ReferenceBookPageStore.isLoaded(cnxId)
+      ReferenceBookPageActions.load(cnxId)
+      MediaStore.once("loaded.#{mediaId}", @updateMedia)
+
+  componentWillUnmount: ->
+    {mediaId} = @props
+    MediaStore.off("loaded.#{mediaId}", @updateMedia)
+
+  updateMedia: (media) ->
+    @setState({media})
+
+  checkShouldPop: ->
+    return true unless @props.mediaDOMOnParent
+    not @isMediaInViewport()
+
+  isMediaInViewport: ->
+    {mediaDOMOnParent, buffer, windowImpl} = @props
+    mediaRect = mediaDOMOnParent.getBoundingClientRect()
+
+    0 <= (mediaRect.top + buffer) <= windowImpl.innerHeight
+
+  highlightMedia: ->
+    {mediaDOMOnParent} = @props
+    mediaDOMOnParent.classList.add('target')
```

Comment:
*Minor:* Can this be something like `link-target`? The term `target` is already used in other parts of the code (but not CSS) to describe other things